### PR TITLE
Dragon's Dogma support

### PR DESCRIPTION
**Pull request by HenryOfCarim**
Experimental support, but some parts can be ported to the main

### CODE REVIEW

File: albam/engines/mtfw/structs/mod-21.ksy [hunk 1]
```diff
@@ -148,10 +157,10 @@ types:
       - {id: face_offset, type: u4}
       - {id: bone_id_start, type: u1}
       - {id: num_weight_bounds, type: u1}
-      - {id: mesh_index, type: u2}
+      - {id: mesh_index, type: u2} # connect_id
```
Comment (Brachi) [inline albam/engines/mtfw/structs/mod-21.ksy:160]: why not change the names now that we know the purpose?

File: albam/engines/mtfw/mesh.py [hunk 2]
```diff
@@ -1409,7 +1431,7 @@ def _export_vertices(app_id, bl_mesh, mesh, mesh_bone_palette, dst_mod, bbox_dat
         vertex_struct.position.x = xyz[0]
         vertex_struct.position.y = xyz[1]
         vertex_struct.position.z = xyz[2]
-        vertex_struct.position.w = 32767  # might be changed later
+        vertex_struct.position.w = 0  # 32767 dd test # might be changed later
```
[the remark this answers is not in the document]
Comment (HenryOfCarim) [reply albam/engines/mtfw/mesh.py:1434]: It wasn't a problem, there was one guess why it crashes DD uses 0 so I tried it and it had 0 effect. maybe need to revert

File: albam/engines/mtfw/structs/mrl.ksy [hunk 3]
```diff
@@ -53,7 +53,8 @@ types:
 
   material:
     seq:
-      - {id: type_hash, type: u4, enum: material_type} # TYPE_nDraw_MaterialStd not in mfx
+      #- {id: type_hash, type: u4, enum: material_type} # TYPE_nDraw_MaterialStd not in mfx
```
[the remark this answers is not in the document]
Comment (HenryOfCarim) [reply albam/engines/mtfw/structs/mrl.ksy:56]: DD uses like 5 material types when I tried to implement it turned out I couldn't just get value like `type_hash` , it always should be like `type_hash._enum_key_` or so, the simplest way to overcome it was to treat it as u4 and use enum on blender side. I am still not sure was it the best solution

File: albam/engines/mtfw/mesh.py [hunk 4]
```diff
@@ -1170,10 +1176,17 @@ def _serialize_meshes_data(bl_obj, bl_meshes, src_mod, dst_mod, materials_map, b
                 p.vertices for p in bl_mesh.data.polygons))
 
         triangles = [e + current_vertex_position for e in triangles]
+        num_indices = len(triangles)
+        # calculate padding for indices
+        if ((num_indices * 2) % 4):
+            triangles.append(triangles[-1])
+            face_padding += 1
+        triangles.append(triangles[-1])
```
[the remark this answers is not in the document]
Comment (Brachi) [reply albam/engines/mtfw/mesh.py:1184]: Ok, turns out yes, the indices are duplicated in the original formats, and we had holes in the format the whole time!
Good finding.

**Review by Brachi — APPROVED** [review]
Bravo!

**Pull request merged**